### ISSUE TITLE
PR is stuck waiting in queue again

### ISSUE THREAD

**Issue by matklad** [bors-ng/bors-ng#206]
One more hanging PR :) 

https://github.com/intellij-rust/intellij-rust/pull/1279

**Comment by notriddle** [comment]
I logged in to the server, and this is what I saw:

```
iex> patch = BorsNG.Database.Repo.get_by!(BorsNG.Database.Patch, project_id: 81, pr_xref: 1279)
%BorsNG.Database.Patch{title: "Typification improvements"}
iex> link = BorsNG.Database.Repo.all(from(p in BorsNG.Database.LinkPatchBatch, where: p.patch_id == ^patch.id))
[]
```

This means that the pull request in question is not assigned to any batches, which implies that it is not in the queue and that it did not run. Weird. The Dashboard agrees with this (it says that the PR is awaiting review).

---

The simplest explanation is that your project's batch management process crashed. When that happens, the supervisor deletes all the running and pending batches (in case one of those caused the crash) then restarts the batch manager. It also makes a best-effort attempt to post a "this batch was canceled" notice to the PR, but if that fails it won't try again.

In this case, I'm pretty sure it wouldn't have worked. You r+-ed that PR at 8 PM UTC on June 5th. That's about a half hour after GitHub reported an outage. I'm guessing (the public bors instance seriously needs better error reporting, because guessing like this sucks) that all of bors's attempts to make API requests to GitHub were timing out, which means that both the attempt to progress that particular batch and the attempt to report an error failed.

**Comment by notriddle** [comment]
TL;DR: r+ it again.

**Comment by ishitatsuyuki** [comment]
@notriddle does it make sense to implement a exponential backoff queue?

**Comment by notriddle** [comment]
Probably in some spots, yeah.

**Comment by matklad** [comment]
@notriddle looks like this is still happens: https://github.com/intellij-rust/intellij-rust/pull/1472. I'll r+ it again now.

**Comment by notriddle** [comment]
Now that #216 is fixed, the next time this happens an error should show up in [intellij-rust/intellij-rust > History](https://bors-app.herokuapp.com/repositories/81/log)
